io/pci/PowerVMEEH.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2017 IBM
+# Author: Venkat Rao B <vrbagal1@linux.vnet.ibm.com>
+
+"""
+This scripts basic EEH tests on all PCI device
+"""
+
+import time
+import sys
+from avocado import main
+from avocado import Test
+from avocado.utils import process
+from avocado.utils import pci
+
+EEH_HIT = 0
+EEH_MISS = 1
+
+
+class EEHRecoveryFailed(Exception):
+    """
+    Exception class, if EEH fails to recover
+    """
+    def __init__(self, thing, dev, log=None):
+        self.thing = thing
+        self.dev = dev
+        self.log = log
+
+    def __str__(self):
+        return "%s %s recovery failed: %s" % (self.thing, self.dev, self.log)
+
+
+class PowerVMEEH(Test):
+
+    """
+    This class contains functions for listing domains
+    forming EEH command
+    """
+    def setUp(self):
+
+        """
+        Gets the console and set-up the machine for test
+        """
+        cmd = "cat /sys/kernel/debug/powerpc/eeh_enable"
+        output = process.system_output(cmd, ignore_status=True, shell=True)
+        if output != '0x1':
+            print "EEH is not enabled, please enable via FSP"
+            sys.exit(1)
+        self.max_freeze = int(self.params.get('max_Freeze', default='1'))
+        cmd = "echo %d > /sys/kernel/debug/powerpc/eeh_max_freezes"\
+            % self.max_freeze
+        process.system(cmd, ignore_status=True, shell=True)
+        self.function = str(self.params.get('function')).split(" ")
+        print "===================Testing EEH Frozen PE======================"
+
+    def test_eeh_basic_pe(self):
+
+        """
+        Test to execute basic error injection on PE
+        """
+        self.pci_address = pci.get_pci_addresses()
+        for self.addr in self.pci_address:
+            self.pci_mem_addr = pci.get_memory_address(self.addr)
+            self.pci_mask = pci.get_mask(self.addr)
+            self.pci_class_name = pci.get_pci_class_name(self.addr)
+            self.pci_interface = pci.get_interfaces_in_pci_address(
+                self.addr, self.pci_class_name)[-1]
+            print "PCI addr = %s" % self.addr
```

Comment:
all can be move in self.log.info and put in two lines or single lines on your preference